prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -5112,6 +5113,111 @@ end subroutine finalizeNegf
+    if (.not. associated(container)) then
+      return
+    end if
+
+    call getChildren(container, "ReferenceOccupation", nodes)
+    nCustomOcc = getLength(nodes)
+    nAtom = size(geo%species)
+    allocate(iAtInRegion(nCustomOcc))
+    allocate(customOcc(orb%mShell, nCustomOcc))
+    allocate(atomOverriden(nAtom))
+    atomOverriden(:) = .false.
+    customOcc(:,:) = 0.0_dp
+    
+    do iCustomOcc = 1, nCustomOcc
+      call getItem1(nodes, iCustomOcc, node)
+      call getChildValue(node, "Atoms", buffer, child=child, multiple=.true.)
+      call convAtomRangeToInt(char(buffer), geo%speciesNames, &
+          & geo%species, child, iAtInRegion(iCustomOcc)%data)
+      if (any(atomOverriden(iAtInRegion(iCustomOcc)%data))) then
+        call detailedError(child, "Atom region contains atom(s) which have&
+            & already been overriden")
+      end if
+      atomOverriden(iAtInRegion(iCustomOcc)%data) = .true.
+      iSpecie = geo%species(iAtInRegion(iCustomOcc)%data(1))
+      if (any(geo%species(iAtInRegion(iCustomOcc)%data) /= iSpecie)) then
+        call detailedError(child, "All atoms in a ReferenceOccupation&
+            & declaration must have the same type.")
+      end if
+      call getOrbitalNames(iSpecie, orb, shellnames)
+      do iShell = 1, orb%nShell(iSpecie)
+          call getChildValue(node, shellnames(iShell), customOcc(iShell, iCustomOcc), &
+            & default=referenceOcc(iShell, iSpecie))
+      end do
+      deallocate(shellnames)
+    end do
+    if (associated(nodes)) then
+      call destroyNodeList(nodes)
+    end if
+
+  end subroutine readCustomReferenceOcc
+
+  !> Builds a unique names for the atomic orbitals 
+  !> Assign 's', 'p', 'd' to first occurring shells, then 's2', 'p2', ... 
+  subroutine getOrbitalNames(iSpecie, orb, shellnames) 
+    !> atomic specie    
+    integer, intent(in) :: iSpecie    
+    !> orbital info
+    type(TOrbitals), intent(in) :: orb
+    !> output string naming the atomic orbital  
+    character(sc), intent(out), allocatable :: shellnames(:)
+
+    integer :: ii, jj 
+    integer, allocatable :: ind(:)
+    !character(sc), allocatable :: names(:)
+    character(1) :: sindx
+
+    !allocate(names(orb%nShell(iSpecie)))
+    allocate(shellnames(orb%nShell(iSpecie)))
+    allocate(ind(orb%nShell(iSpecie)))
+    ind = 1 
+
+    do ii = 1, orb%nShell(iSpecie) 
+      write(shellnames(ii), "(A)") orbitalNames(orb%angShell(ii, iSpecie) + 1)
+      if (any(shellnames(ii) == shellnames(1:ii-1))) then
+        ind(ii) = ind(ii) + 1
+        select case (ind(ii))
+        case(1)
+          sindx = ""
+        case(2:9)
+          write(sindx,'(I1)') ind(ii)   
+        case(10:)
+          call error("Exceeded maximum number of shells with the same angular momentum (9)")     
+        end select 
+        shellnames(ii) = trim(adjustl(shellnames(ii)))//sindx
+      end if
+      print*,shellnames(ii)
```

Comment:
Should this print be here? Probably initprogram should be printing this information as part of the custom occupation block, also StdOut unit instead of *.